pyrit/datasets/fetch_examples.py

```diff
@@ -0,0 +1,155 @@
+            return FILE_TYPE_HANDLERS[file_type]["read"](file)
+        else:
+            raise ValueError("Invalid file_type. Expected 'json', 'csv', or 'txt'.")
+
+
+def _write_cache(cache_file: Path, examples: List[Dict[str, str]], file_type: str):
+    """
+    Write data to cache.
+    """
+    cache_file.parent.mkdir(parents=True, exist_ok=True)
+    with cache_file.open("w") as file:
+        if file_type in FILE_TYPE_HANDLERS:
+            FILE_TYPE_HANDLERS[file_type]["write"](file, examples)
+        else:
+            raise ValueError("Invalid file_type. Expected 'json', 'csv', or 'txt'.")
+
+
+def _fetch_from_public_url(source: str, file_type: str) -> List[Dict[str, str]]:
+    """
+    Fetch examples from a repository.
+    """
+    response = requests.get(source)
+    if response.status_code == 200:
+        if file_type in FILE_TYPE_HANDLERS:
+            if file_type == "json":
+                return FILE_TYPE_HANDLERS[file_type]["read"](io.StringIO(response.text))
+            else:
+                return FILE_TYPE_HANDLERS[file_type]["read"](
+                    io.StringIO("\n".join(response.text.splitlines()))
+                )  # noqa: E501
+        else:
+            raise ValueError("Invalid file_type. Expected 'json', 'csv', or 'txt'.")
+    else:
+        raise Exception(f"Failed to fetch examples from public URL. Status code: {response.status_code}")
+
+
+def _fetch_from_file(source: str, file_type: str) -> List[Dict[str, str]]:
+    """
+    Fetch examples from a local file.
+    """
+    with open(source, "r") as file:
+        if file_type in FILE_TYPE_HANDLERS:
+            return FILE_TYPE_HANDLERS[file_type]["read"](file)
+        else:
+            raise ValueError("Invalid file_type. Expected 'json', 'csv', or 'txt'.")
+
+
+def fetch_examples(
+    source: str,
+    source_type: Literal["public_url", "file"] = "public_url",
+    cache: bool = True,
+    data_home: Optional[Path] = None,
+) -> List[Dict[str, str]]:
+    """
+    Fetch examples from a specified source with caching support.
+
+    Args:
+        source (str): The source from which to fetch examples.
+        source_type (Literal["public_url", "file"]): The type of source ('public_url' or 'file'). Defaults to 'public_url'.
+        file_type (str): The type of file ('json', 'csv', or 'txt'). Defaults to 'json'.
+        cache (bool): Whether to cache the fetched examples. Defaults to True.
+        data_home (Optional[Path]): Directory to store cached data. Defaults to None.
+
+    Returns:
+        List[Dict[str, str]]: A list of examples.
+
+    Example usage:
+        examples = fetch_examples(
+            source='https://raw.githubusercontent.com/KutalVolkan/many-shot-jailbreaking-dataset/5eac855/examples.json',
+            source_type='public_url'
+        )
+    """
+
+    file_type = source.split(".")[-1]
+    if file_type not in FILE_TYPE_HANDLERS:
+        raise ValueError("Invalid file_type. Expected 'json', 'csv', or 'txt'.")
```

Comment:
When a new file_type is added, error message will be taken care automatically rather than updating the error message when a new file_type is added
```suggestion
            valid_types = ", ".join(FILE_TYPE_HANDLERS.keys())
            raise ValueError(f"Invalid file_type. Expected one of: {valid_types}.")
```